libraries/botbuilder-dialogs/src/prompts/oauthPrompt.ts

```diff
@@ -315,6 +339,40 @@ export class OAuthPrompt extends Dialog {
             {
                 await context.sendActivity({ type: 'invokeResponse', value: { status: 500 }});
             }
+        } else if (this.isTokenExchangeRequestInvoke(context)) {
```

Comment:
This logic below (all the if's), require some comments (one per case ideally) explaning what it means to be in each of these states